plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -292,10 +297,21 @@ func (s *serviceAccount) limitSecretReferences(serviceAccount *api.ServiceAccoun
 			continue
 		}
 		secretName := source.Secret.SecretName
-		if !referencedSecrets.Has(secretName) {
+		if !mountableSecrets.Has(secretName) {
```

Comment:
I'm not sure-- it seems weird to me that I can make secret A, assign it to service account B, make my pod *not* use any service account, and then still use secret A.

Also considering that all pods get the default service account, doesn't this mean you have to put refs to your secrets into the default service account before you can use them?